### PR TITLE
feat(tutorial): add step 2 instructions

### PR DESCRIPTION
**Pull request by jnm2377**
wip, but mostly done. Need to address some comments and then test it out.

Netlify url: https://deploy-preview-1601--carbon-website.netlify.com/tutorial/react-step-2/
(tutorial is not currently in sidebar nav)

### CODE REVIEW

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+  forwardText="Next page"
+  pageSize="{100}"
+  pageSizes="{[10,"
+  25,
+  50,
+  100]}
+  itemsPerPageText="Items per page"
+/>
+```
+
```
Comment (asudoh) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Good to add a note that application should (only) render rows corresponding to the current page from `<Pagination>`.

Comment (jnm2377) [reply src/content/tutorial/react-step-2/react-step-2.mdx]: Thanks for the feedback. That's a good point. I'll make sure to add it. 🎉

**Comment by tw15egan** [comment]
Going to run through this today 👍

**Comment by jnm2377** [comment]
@tw15egan I still need to add some things, since we made some changes to what we built 👀 will hopefully be done by tomorrow morning (I have a few meetings the rest of the day)

**Comment by tw15egan** [comment]
👍 Sounds good! Just let me know when it's ready to be tested

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+    </TableHead>
+    <TableBody>
+      {rows.map(row => (
+      <React.Fragment key="{row.id}">
+        <TableExpandRow {...getRowProps({ row })}>
+          {row.cells.map(cell => (
+          <TableCell key="{cell.id}">{cell.value}</TableCell>
+          ))}
+        </TableExpandRow>
+        <TableExpandedRow colSpan="{headers.length" + 1}>
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: should be `colSpan={headers.length + 1}`

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+        </TableHeader>
+        ))}
+      </TableRow>
+    </TableHead>
+    <TableBody>
+      {rows.map(row => (
+      <React.Fragment key="{row.id}">
+        <TableExpandRow {...getRowProps({ row })}>
+          {row.cells.map(cell => (
+          <TableCell key="{cell.id}">{cell.value}</TableCell>
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Shouldn't have quotes

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+        {headers.map(header => (
+        <TableHeader {...getHeaderProps({ header })}>
+          {header.header}
+        </TableHeader>
+        ))}
+      </TableRow>
+    </TableHead>
+    <TableBody>
+      {rows.map(row => (
+      <React.Fragment key="{row.id}">
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Shouldn't have quotes

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+
+#### Pagination
+
+Last but not least, we'll need to add our pagination component as a sibling of `<DataTable/>`.
+
+```javascript
+<DataTable render={({ rows, headers, getHeaderProps, getRowProps, getTableProps,
+}) => ( ... )} />
+<Pagination
+  totalItems="{1228}"
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Shouldn't have quotes

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+  backwardText="Previous page"
+  forwardText="Next page"
+  pageSize="{100}"
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Shouldn't have quotes

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 7]
```diff
@@ -0,0 +1,641 @@
...
+  pageSizes="{[10,"
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: Shouldn't have quotes

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 8]
```diff
@@ -0,0 +1,641 @@
...
+```
+
+Congratulations! We've now created our static repo page!
+
+## Submit pull request
+
+We have now built our landing and repo page, so let's run our tests again.
+
+```bash
+$ yarn test
```
Comment (tw15egan) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: I still get the `ariaLabel` error 🤔

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 9]
```diff
@@ -0,0 +1,641 @@
...
+  Tab,
+} from 'carbon-components-react';
+```
+
+Before we can render the tabs, we need to add some props for the component after the import.
+
+```javascript
+const props = {
+  tabs: () => ({
+    className: 'some-class',
```
Comment (asudoh) [inline src/content/tutorial/react-step-2/react-step-2.mdx]: I'm imagining that you ported the code from our Storybook code. Please note that the props being function there is the nature of Storybook knobs. For props in applications props here can be simple key-map values.

Update: Or you can directly put the props to below React rendering code.

File: src/content/tutorial/react-step-2/react-step-2.mdx (hunk 5 above)
Comment (mattrosno) [reply src/content/tutorial/react-step-2/react-step-2.mdx]: Might need to wrap this in Prettier ignore comments. I'm guess Prettier is adding the quotes.

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 10]
```diff
@@ -0,0 +1,641 @@
...
+  TableExpandedRow,
+} from 'carbon-components-react';
+```
+
+#### Data table props
+
+Before we can add the components to our `return` section, we need to create arrays for our row content and our header content, which will live in our `render` section.
+
+```javascript
+export default class RepoPage extends Component {
```
[the remark this answers is not in the document]
Comment (jnm2377) [reply src/content/tutorial/react-step-2/react-step-2.mdx]: Thanks for catching this. 👍

File: src/content/tutorial/react-step-2/react-step-2.mdx (hunk 9 above)
Comment (jnm2377) [reply src/content/tutorial/react-step-2/react-step-2.mdx]: Thanks for the explanation! I updated the code! 👍

File: src/content/tutorial/react-step-2/react-step-2.mdx [hunk 11]
```diff
@@ -0,0 +1,632 @@
...
+`grid-columns-16: true` in our `$feature-flags`.
+
+```css
+$feature-flags: (
+  ui-shell: true,
+  grid-columns-16: true
+);
+```
+
+Before we start our development server, run `yarn test`. When prompted for a watch mode, you can press `a` to run all tests. You'll notice that there are some tests failing. Just like in step one, we'll make these tests pass as we build our app.
```
[the remark this answers is not in the document]
Comment (jnm2377) [reply src/content/tutorial/react-step-2/react-step-2.mdx]: Yeah I think the `aria-label` was a warning. I added an `aria-label` to the `HeaderGlobalActions`/instructions so it'll go away. 👍 Thanks for the catch!

**Pull request merged**